src/io/file_descriptor.cr

```diff
@@ -72,29 +72,29 @@ class IO::FileDescriptor < IO
   # This might be different from the internal file descriptor. For example, when
   # `STDIN` is a terminal on Windows, this returns `false` since the underlying
   # blocking reads are done on a completely separate thread.
-  def blocking
+  def blocking : Bool
     emulated = emulated_blocking?
     return emulated unless emulated.nil?
     system_blocking?
   end
 
-  def blocking=(value)
+  def blocking=(value : Bool) : Int32?
     self.system_blocking = value
   end
```

Comment:
These combination of input and return types should not be correct. The return type of an assignment should be the value assigned. Does `system_blocking=` do anything weird?

Reply:
`system_blocking=` may leak the return value of `fcntl`.
We should probably return `value` (c.f. #10083).

For this PR, I'd suggest to just set the return type restriction to `Nil`.

Reply:
Per #10083, I could also update this method to return `Bool` and put `value` at the end?

Reply:
Yes, it's best to keep this PR focused on adding type restrictions, without changing anything else.